### PR TITLE
Add Hbase System Logging Receiver

### PR DESCRIPTION
**Pull request by Mrod1598**
Example logs:
image

### CODE REVIEW

**Comment by Mrod1598** [comment]
Hbase passes, failures are related to tomcat.

File: docs/hbase.md
Comment (qingling128) [inline docs/hbase.md:12]: Also `wildcard_refresh_interval`?

e.g.

```
| `wildcard_refresh_interval` | `60s` | The interval at which wildcard file paths in `include_paths` are refreshed. Specified as a time interval parsable by [time.ParseDuration](https://pkg.go.dev/time#ParseDuration). Must be a multiple of 1s.|
```

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (qingling128) [inline integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:7]: Chances are a bit higher that users will use the default package manager (e.g. yum, apt-get, zyypper) to install a package instead of a pinned version `2.4.9`. Can we adjust the `install` script to reflect that?

File: integration_test/third_party_apps_data/applications/hbase/sles/post
Comment (qingling128) [inline integration_test/third_party_apps_data/applications/hbase/sles/post:3]: Looks like the `submodules/fluent-bit` change below got included by accident. Can we remove it from the change set?

**Review by qingling128 — COMMENTED** [review]
Left some minor comments.

File: integration_test/third_party_apps_data/applications/hbase/sles/post
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/sles/post:3]: Removed

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:7]: Same as with hbase-metrics where there doesn't appear to be a package

Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:7]: the [official docs ](https://hbase.apache.org/book.html#quickstart)also point to using a tar

File: docs/hbase.md
Comment (Mrod1598) [reply docs/hbase.md:13]: Added

File: integration_test/third_party_apps_data/agent/ops-agent/linux/enable_hbase
Comment (quentinmit) [inline integration_test/third_party_apps_data/agent/ops-agent/linux/enable_hbase:26]: Nit: Name the sample pipeline `hbase_system` to match the receiver.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:7]: I don't see any distro packaging for hbase either.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [inline integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:12]: I looked through the docs and the release tarball and I can't for the life of me find a single set of instructions for how you are supposed to start hbase on boot. Where did you find this systemd unit? Can you add a link in a comment?

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [inline integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:18]: It's probably not important if this is reliable enough for a test, but AFAICT `start-hbase.sh` will start multiple daemons, but `Type=forking` expects only a single master process, so it's possible that systemd will be confused and `stop`/`restart` will misbehave.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [inline integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:29]: Unnecessary use of `cat`. This can just be `sudo tee /opt/hbase/conf/hbase-env.sh <<EOF`.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [inline integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:31]: You have port 9999 but the sample config in the upstream `hbase-env.sh` sets 10101; we should use the default port.

Also, the comments in the sample config recommend using a different mechanism to configure JMX: https://hbase.apache.org/book.html#jmx_config

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [inline integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:36]: `systemctl enable --now hbase` will also start it.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [inline integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:30]: It's not important in this PR because this is just logs, but we'll need to think in the metrics PR about whether this is sufficient - I believe this only configures hbase in standalone mode, not distributed mode.

File: docs/hbase.md
Comment (qingling128) [reply docs/hbase.md:13]: Thanks!

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (qingling128) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:7]: Ah I see. NVM

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:30]: Ok, we will address this on the metrics PR.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:12]: Apache doesn't provide a service file, we based ours off of [this](https://github.com/esamson/baseboxes/blob/master/sl-tools/etc/systemd/system/hbase.service).

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:31]: [Here](https://hbase.apache.org/book.html#_get_started_with_hbase) it also discusses setting up the environment variables in this [file](https://github.com/apache/hbase/blob/master/conf/hbase-env.sh) which already has commented out options in it. it seemed cleaner to do it this way. Also, most of the top results from a quick google search shows using environment variables. So, if we're trying to replicate what most users would do I think this would be it.

File: integration_test/third_party_apps_data/agent/ops-agent/linux/enable_hbase
Comment (Mrod1598) [reply integration_test/third_party_apps_data/agent/ops-agent/linux/enable_hbase]: Updated

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install]: Updated

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install]: Updated

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (Mrod1598) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install]: removed the service file and just using the env file as we only need to start it and stop it.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install:12]: Yeah, I don't know which of the two ways to configure JMX is better, but I do know you're not using the default port here - why not?

(The file you linked contains:
```
# NOTE: HBase provides an alternative JMX implementation to fix the random ports issue, please see JMX
# section in HBase Reference Guide for instructions.
```
That's what I was referencing with the link to #jmx_config)

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (qingling128) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install]: Thanks!

**Review by qingling128 — APPROVED** [review]
LGTM modulo the open question about default port is addressed and integration tests are fixed (Probably just minor issues):

image

**Review by qingling128 — APPROVED** [review]
LGTM

The remaining open threads do not seem like blockers for this PR (In case some improvements are preferred, we can send follow up PRs).

Waiting for the integration tests for a final check before merging.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install]: I defer to your HBase knowledge. I've never used HBase so I don't know if one of the JMX configurations is more popular than the other. At some point it would be great if we could test both, but we don't have a good way to do that right now.

File: integration_test/third_party_apps_data/applications/hbase/centos_rhel/install
Comment (quentinmit) [reply integration_test/third_party_apps_data/applications/hbase/centos_rhel/install]: Interesting. I will be very happy when we move on to non-Java software that has more standard installation/configuration mechanisms!

**Comment by qingling128** [comment]
hbase tests seem to be failing:

```
--- FAIL: TestThirdPartyApps (0.00s)
    --- FAIL: TestThirdPartyApps/debian-10 (0.00s)
        --- SKIP: TestThirdPartyApps/debian-10/mysql (0.00s)
        --- PASS: TestThirdPartyApps/debian-10/memcached (113.75s)
        --- PASS: TestThirdPartyApps/debian-10/redis (119.03s)
        --- PASS: TestThirdPartyApps/debian-10/mongodb (119.08s)
        --- PASS: TestThirdPartyApps/debian-10/apache (137.27s)
        --- PASS: TestThirdPartyApps/debian-10/postgresql (138.94s)
        --- PASS: TestThirdPartyApps/debian-10/nginx (153.03s)
        --- PASS: TestThirdPartyApps/debian-10/couchdb (158.45s)
        --- PASS: TestThirdPartyApps/debian-10/jvm (166.41s)
        --- PASS: TestThirdPartyApps/debian-10/elasticsearch (175.78s)
        --- PASS: TestThirdPartyApps/debian-10/cassandra (178.73s)
        --- PASS: TestThirdPartyApps/debian-10/solr (225.06s)
        --- FAIL: TestThirdPartyApps/debian-10/tomcat (248.54s)
        --- FAIL: TestThirdPartyApps/debian-10/hbase (614.81s)

--- FAIL: TestThirdPartyApps (0.00s)
    --- FAIL: TestThirdPartyApps/sles-12 (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/apache (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/redis (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/couchdb (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/jvm (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/cassandra (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/mysql (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/nginx (0.00s)
        --- PASS: TestThirdPartyApps/sles-12/mongodb (249.66s)
        --- PASS: TestThirdPartyApps/sles-12/postgresql (257.86s)
        --- PASS: TestThirdPartyApps/sles-12/elasticsearch (264.22s)
        --- PASS: TestThirdPartyApps/sles-12/memcached (271.52s)
        --- PASS: TestThirdPartyApps/sles-12/solr (321.72s)
        --- PASS: TestThirdPartyApps/sles-12/tomcat (339.55s)
        --- FAIL: TestThirdPartyApps/sles-12/hbase (790.46s)

--- FAIL: TestThirdPartyApps (0.00s)
    --- FAIL: TestThirdPartyApps/sles-15 (0.00s)
        --- SKIP: TestThirdPartyApps/sles-15/apache (0.00s)
        --- SKIP: TestThirdPartyApps/sles-15/nginx (0.00s)
        --- SKIP: TestThirdPartyApps/sles-15/mysql (0.00s)
        --- SKIP: TestThirdPartyApps/sles-15/couchdb (0.00s)
        --- SKIP: TestThirdPartyApps/sles-15/cassandra (0.00s)
        --- SKIP: TestThirdPartyApps/sles-15/redis (0.00s)
        --- SKIP: TestThirdPartyApps/sles-15/jvm (0.00s)
        --- PASS: TestThirdPartyApps/sles-15/mongodb (297.69s)
        --- PASS: TestThirdPartyApps/sles-15/memcached (305.94s)
        --- PASS: TestThirdPartyApps/sles-15/solr (333.20s)
        --- PASS: TestThirdPartyApps/sles-15/elasticsearch (337.60s)
        --- PASS: TestThirdPartyApps/sles-15/tomcat (355.72s)
        --- PASS: TestThirdPartyApps/sles-15/postgresql (374.96s)
        --- FAIL: TestThirdPartyApps/sles-15/hbase (775.06s)

--- FAIL: TestThirdPartyApps (0.00s)
    --- FAIL: TestThirdPartyApps/ubuntu-2004-lts (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/mysql (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/nginx (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/apache (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/cassandra (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/redis (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/jvm (0.00s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/postgresql (181.92s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/memcached (188.25s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/mongodb (194.42s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/couchdb (211.90s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/elasticsearch (225.67s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/tomcat (261.59s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/solr (280.40s)
        --- FAIL: TestThirdPartyApps/ubuntu-2004-lts/hbase (719.28s)
```

**Comment by qingling128** [comment]
Just checking in, this one is still WIP, right (Looks like the last integration tests had some failures)? Trying to figure out whether it's waiting for review.

image

**Review by qingling128 — APPROVED** [review]
LGTM

Hbase tests passed:

```
--- FAIL: TestThirdPartyApps (0.00s)
    --- FAIL: TestThirdPartyApps/sles-12 (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/jvm (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/apache (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/mysql (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/nginx (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/cassandra (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/couchdb (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/redis (0.00s)
        --- SKIP: TestThirdPartyApps/sles-12/wildfly (0.00s)
        --- PASS: TestThirdPartyApps/sles-12/mongodb (984.19s)
        --- PASS: TestThirdPartyApps/sles-12/activemq (1044.26s)
        --- PASS: TestThirdPartyApps/sles-12/elasticsearch (1089.34s)
        --- PASS: TestThirdPartyApps/sles-12/tomcat (1103.43s)
        --- PASS: TestThirdPartyApps/sles-12/postgresql (1118.50s)
        --- PASS: TestThirdPartyApps/sles-12/memcached (1169.51s)
        --- PASS: TestThirdPartyApps/sles-12/hbase (1723.09s)
        --- PASS: TestThirdPartyApps/sles-12/zookeeper (1759.07s)
        --- PASS: TestThirdPartyApps/sles-12/solr (1820.92s)
        --- FAIL: TestThirdPartyApps/sles-12/kafka (7227.54s)

--- FAIL: TestThirdPartyApps (0.00s)
    --- FAIL: TestThirdPartyApps/ubuntu-2004-lts (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/cassandra (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/apache (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/jvm (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/wildfly (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/redis (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/mysql (0.00s)
        --- SKIP: TestThirdPartyApps/ubuntu-2004-lts/nginx (0.00s)
        --- FAIL: TestThirdPartyApps/ubuntu-2004-lts/zookeeper (3.24s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/mongodb (219.69s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/hbase (220.01s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/couchdb (221.26s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/postgresql (229.57s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/memcached (238.02s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/kafka (247.56s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/elasticsearch (270.41s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/activemq (271.81s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/solr (280.25s)
        --- PASS: TestThirdPartyApps/ubuntu-2004-lts/tomcat (346.20s)
```

Can be merged after rebasing.

**Pull request merged**
